File: fun.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Print("Enter the first number: ")
	num1Str, _ := reader.ReadString('\n')
	num1, err := strconv.ParseFloat(strings.TrimSpace(num1Str), 64)
	if err != nil {
		fmt.Println("Invalid number.")
		return
	}

	fmt.Print("Enter an operator (+, -, *, /): ")
	opStr, _ := reader.ReadString('\n')
	operator := strings.TrimSpace(opStr)

	fmt.Print("Enter the second number: ")
	num2Str, _ := reader.ReadString('\n')
	num2, err := strconv.ParseFloat(strings.TrimSpace(num2Str), 64)
	if err != nil {
		fmt.Println("Invalid number.")
		return
	}

	var result float64
	switch operator {
	case "+":
		result = num1 + num2
	case "-":
		result = num1 - num2
	case "*":
		result = num1 * num2
	case "/":
		if num2 == 0 {
			fmt.Println("Error: Cannot divide by zero.")
			return
		}
		result = num1 / num2
	default:
		fmt.Println("Invalid operator.")
		return
	}

	fmt.Printf("Result: %f\n", result)
}

File: fun.pl
use strict;
use warnings;

print "Enter the first number: ";
my $num1 = <STDIN>;
chomp $num1;

print "Enter an operator (+, -, *, /): ";
my $operator = <STDIN>;
chomp $operator;

print "Enter the second number: ";
my $num2 = <STDIN>;
chomp $num2;

my $result;

if ($operator eq '+') {
    $result = $num1 + $num2;
} elsif ($operator eq '-') {
    $result = $num1 - $num2;
} elsif ($operator eq '*') {
    $result = $num1 * $num2;
} elsif ($operator eq '/') {
    if ($num2 == 0) {
        die "Error: Cannot divide by zero.\n";
    }
    $result = $num1 / $num2;
} else {
    die "Error: Invalid operator.\n";
}

print "Result: $result\n";
